schemachange/sc_add_table.c

```diff
@@ -228,8 +228,17 @@ int do_add_table(struct ireq *iq, struct schema_change_type *s,
     struct dbtable *db;
     set_empty_options(s);
 
-    if ((rc = check_option_coherency(s, NULL, NULL))) return rc;
+    if(!s->is_osql){
```

Comment:
I am thinking if we should drop this condition, so that we always check whether a time-partition view with the same name exists.

Reply:
We also need a similar check in do_dryrun().

Reply:
ddl is already protected, so I would leave this only for legacy. Modern ddl is called for multi-sc transactions, some of which might get the lock already, at which point this will deadlock